[team-05] iOS IssueListView와 IssueFilterView 구현

Aiden0704 opened the pull request:
# ⭐️ 주요 작업

### 이슈 목록 CollectionView 구현

- [x]  issueCell에 들어갈 제목, 설명, 마일스톤, 레이블 구현
- [x]  각각의 Constraints 설정
- [x]  IssueList에서 Header로 사용할 UICollectionReusableView 구현
- [x]  IssueList의 UINavigationController 구현

### 이슈 필터 CollectionView 구현

- [x]  issueFilter 목록을 나타내는 UIViewController 구현
- [x]  issueFilter 목록에서 셀에 들어갈 UICollectionViewCell 구현
- [x]  issueFilter 목록에서 헤더에 들어갈 UICollectionReusableView 구현

### 구현에 필요한 font, color, SFSymbol 타입 구현

- [x]  Font를 UIFont를 담고있는 객체 구현
- [x]  Color값을 담고있는 객체 구현
- [x]  SFSymbol 이름을 담고있는 객체 구현

# ⭐️ 스크린 샷 또는 GIF
![Simulator_Screen_Recording_-_iPhone_14_Pro_-_2023-05-19_at_11 19 08](https://github.com/codesquad-members-2023/issue-tracker/assets/115064144/61c6eec2-332e-43ac-b4ab-a7853c5bb5b4)

# ⭐️ 학습한 부분

- navigationController
- constraint

# ⭐️ 고민한 부분과 해결 과정

## 1. 컬러, 폰트, 심볼을 어디까지 타입으로 묶어야 하는가?

- 현재 컬러 팔래트, 폰트스타일, SF심볼을 타입으로 묶어 코드 작성시 오타의 확률을 줄임

## 2. Modal에서 navigationBar를 띄우는 방법

- issueListController를 navigationController의 root로 설정한뒤 navigationController의 self.present(issueFilterView, animation: true)를 사용해 modal을 띄움
- 이때 modal의 navigationBar가 사라지는 현상이 발생하여 modal로 navigationBar를 띄우는 방법을 고민
- issueFilterView를 navigationController로 다시한번 감싸 issueFilterView를 root로 만들어주는 방법 체택
    
    ### **질문?**
    
    **이때, issueFilterView를 navigationController로 만들면 backBarButtonItem을 사용할 수 없고, leftBarButtonItem를 사용해 기능구현을 해줘야하는 불편함이 있고. naviagationController안에 또 navigationController가 있는 형태여서 이게 맞는 구조인가라는 의문이 있습니다.**
    

## 3. 뷰에서도 할 수 있다면, 인터페이스로 추상화를 하는게 일반적인가?

- 현재 프로젝트에서는 제목, 설명, 마일스톤, 레이블을 생성할 때, 공통되는 부분이 있다고 생각하여 인스턴스 형태로 생성을 하게 구현하였습니다
- 그래야 나중에 추가를 할 때, 더 유용하고 손 쉽게 확장 및 개선이 가능하다고 생각했기 때문입니다
    
    ### **질문?**
    
    **여기서 궁금한 부분이 공통되는 부분을 프로토콜로 추상화를 하고, 해당 프로토콜의 타입으로 제목, 설명, 마일스톤, 레이블을 생성하면 어떨까라는 생각을 하게 되었습니다**
    
    **ex) func createText(of text: String) → String**
    
    **프로토콜의 추상화로 인해 객체간의 의존성이 줄어들고, 프로토콜만의 의존성이 생기니 이게 더 좋은 방법이겠지? 라는 생각을 하였지만, 프로토콜의 추상화를 뷰 부분에서도 적용을 하나? 라는 생각을 하였습니다**
    
    **아직 확신이 들지않았고 혹시라도 잘못 알고있는 부분과 놓치는 부분이 있을까 라는 생각 때문에 이번 PR때는 적용하지 않았습니다**
    
    **이 부분에 대하여 지이의 의견을 듣고 싶습니다! 그리고 혹시 많이 찾아보았지만 참고할만한 글이 있다면 추천 부탁드립니다!**
    

## 4. StackView를 써야하는가?

- 설명에서 가변적으로 1줄~2줄이 들어감
- stackView의 각 height가 정해져 있어 설명이 2줄이 되었을떄 가변적으로 늘어나지 않음
- 따라서, 각 라벨에 constraint를 따로 주는 방식으로 작성
    
    ### **질문?**
    
    **cell에서 label이 쌓여잇을때 stackView로 묶어 부모에서 constraint를 관리하는게 선호되는 방식인지, stackview로 묶지않고 label단위로 관리하는게 선호되는 방식인지 궁금합니다.**
    

## TODO

- 이번 PR에는 DataSource를 사용하였지만, 다음 PR때는 Diffable DataSource로 개선
- 현재 Dummy Data를 사용하지만, 다음 PR에는 서버와 연결

Aiden0704 committed:
chore: ColorValue 추가

- ColorValue 파일 생성

HansolWorld committed:
chore: init Font Style

Aiden0704 committed:
feat: 피드백 반영

- title과 image를 enum으로 재구성 후, 피드백에 알맞게 수정

Aiden0704 committed:
feat: TabBarController 피드백 반영

- enum으로 구현한 image와 title을 TabBarType으로 접근하여 사용
- 해당 클래스는 상속이 이루어질 일이 없다고 판단하여 final 키워드 추가

HansolWorld committed:
feat: Font 객체 생성

HansolWorld committed:
feat: static let 적용

Aiden0704 committed:
chore: ColorValue enum 생성 및 설정

- 전역으로 사용 가능한 color value 설정

HansolWorld committed:
feat: issue Filter를 구현하기 위한 초기 설정
- Issue Filter CollectionView Controller 생성
- Issue Filter Collection View Cell 생성

HansolWorld committed:
feat: 사용하는 Font에 대한 타입 설정

HansolWorld committed:
view: IssueFilter에 들어가는 cell 구현

HansolWorld committed:
view: IssueFilterView에 들어가는 header 정의

Aiden0704 committed:
feat: CustomLabelView 구현

- 재사용이 가능한 CustomLabelView 구현

Aiden0704 committed:
view: 이슈 목록 Cell 구현

- Cell위에 올라가는 Label 구현 및 Constraints 설정

HansolWorld committed:
view: IssueFilterView 구현
- 이슈목록에서 필터 클릭시 나오는 view구현
- issueFilterHeader와 issueFilterCell 연결

Aiden0704 committed:
view: HeaderView 구현

- Section을 나타내는 HeaderView 구현

Aiden0704 committed:
view: IssueViewController 구현

- 이슈 목록에 대한 Controller 구현 및 Constraints 설정
- CollectionView를 표시하기 위한 DataSource, Delegate 구현

Aiden0704 committed:
feat: SFsymbol enum 구현

- 공통적으로 사용할 SFsymbol을 담은 enum 구현

Aiden0704 committed:
rename: SymbolImage 파일 이름 변경

- SymbolImage -> TabBarType

Aiden0704 committed:
view: TabBarViewController 수정

- NavigationController 구현 및 BarItem 추가
- NavigationController rootViewController 설정

Aiden0704 committed:
chore: 파일 추가

Aiden0704 committed:
Merge branch 'dev-ios' into feat/ios/init_issue_list_collection_view

HansolWorld committed:
Merge pull request #76 from codesquad-members-2023-group5/feat/ios/init_issue_list_collection_view

[iOS] 이슈 목록 CollectionView 구현

HansolWorld committed:
Merge branch 'dev-ios' into feat/ios/issue-filter-view

HansolWorld committed:
Merge pull request #75 from codesquad-members-2023-group5/feat/ios/issue-filter-view

[iOS] Feat: issue filter view setting

HansolWorld committed:
feat: FontStyle 타입 사용
- systemFont코드를 FontStyle을 사용하게 적용

HansolWorld committed:
feat: SFSymbol 추가

HansolWorld committed:
feat: issuFilterView에 SFsymbol 타입 적용
- SFsymbol 타입을 사용해 UIImage생성

HansolWorld committed:
view: issue목록과 issue필터 연결
- modal을 사용해 두 view 연결
- issueFilterView에 UINavigationController 설정

HansolWorld committed:
view: tabbar background 색상 변경

Aiden0704 committed:
style: 코드 스타일 일치

- 페어와 코드 스타일 일치

Aiden0704 committed:
style: 코드 스타일 개선 및 코드 분리

- 페어와 코드 스타일 일치
- 메소드 분리

Aiden0704 committed:
Chore: MockHeaderData 파일 생성

- 이슈 필터 목록의 Header에 해당하는 DataFile 생성

Aiden0704 committed:
feat: Dummy Data 적용 및 코드 스타일 개선

- configureTitle() 메소드 구현
- 페어와 코드 스타일 일치

Jeeehee commented:
Aiden과 Sol 이번주도 애쓰셨습니다! ☺️
리뷰를 읽으시다 이해 안가시는 부분이 있다면 말씀해주세요 ㅎ_ㅎ

먼저, 많은 고민을 하신게 느껴져 참 보기 좋습니다!
다만 두 분이 각자 구현하신 IssueViewController와 IssueFilterViewController + 연결 된 HeaderView까지 구현하신 스타일이 많이 다릅니다.
함께 하는 프로젝트이니 많은 소통과 함께, 서로의 코드를 잘 살펴보고 맞춰가면 좋겠습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/CommonTypeValue/FontStyle.swift:
```diff
@@ -0,0 +1,83 @@
+}
+      
+struct Factor {
+    let font: UIFont
```
아래 lineLength는 사용되지 않는 것 같은데 왜 선언해두신걸까요?

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueFilter/IssueFilterCell.swift:
```diff
@@ -0,0 +1,90 @@
+
+final class IssueFilterCell: UICollectionViewCell {
+    static let identifier = "IssueFilterViewCell"
+    var isChecked = false
```
다른 View에서 사용하지 않는 것 같은데, 해당 변수가 private이 아닌 이유가 있을까요?
또 UICollectionViewCell에는 isSeleted라는 속성을 제공해주고 있습니다!
한 번 찾아보시고 사용해보셔도 좋을 것 같네요!

Aiden0704 replied:
isSelected를 사용했습니다!
사용함으로써 didSelectItemAt, updateCheckmarkState 메소드를 삭제했습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueFilter/IssueFilterCell.swift:
```diff
@@ -0,0 +1,90 @@
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
```
여기부터 하단까지 쭉 View를 구성하는 메서드인데, 각각 분리되어 있어 코드가 불필요하게 길어지는 것 같습니다.
모두 View를 구성하거나, View의 Layout을 잡아주는 것으로 성질이 같으니 하나 혹은 두개의 메서드로 묶어보면 어떨까요?

Aiden0704 replied:
말씀해주신 방향대로 수정하였습니다!

IssueFilterCell에서 각자의 label마다 constraint를 다른 메서드에서 관리하면 나중에 관련된 constraint를 찾기 편하지 않을까 하는 생각을 했습니다.
그런데 생각해보니 label이 다른 label과 연관되어 constraint를 설정하는거라 하나의 메서드로 관리하는게 맞을거란 생각을 했습니다.

IssueCardCell에서 하나의 메서드에서 너무 많은 constraint를 관리하게되서 메서드가 무거워지는게 아닌가하는 걱정과, 이런 이유에서 lint에 func_body_length를 25줄이 넘어가지 안게 설정했었는데. 혹시 어떻게 생각하시는지 궁금합니다!

Jeeehee replied:
분리하신 의도는 너무 좋습니다! 
근데 말씀하신대로, AutoLayout을 구현하게 되면 서로의 Constraints를 알아야 하는 경우가 많죠!
View가 복잡하지 않다면, 하나의 메서드에서 처리해주는게 가독성이나 유지보수 면에서 좋다고 생각합니다!
단순히 Line의 길이가 길어진다하여 메서드가 무거워 진다고 생각하진 않습니다! 🙂

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueFilter/IssueFilterCell.swift:
```diff
@@ -0,0 +1,90 @@
+        ])
+    }
+    
+    func updateCheckmarkState() {
```
해당 로직은 아래와 같이 짜면, 코드의 길이도 줄어들고 가독성도 높일 수 있습니다!

```swift
let checkImageColor = isChecked ? ColorValue.gray700 : ColorValue.blue
checkImage.tintColor = checkImageColor
isChecked.toggle()
```

Aiden0704 replied:
저희가 그라운드룰을 정할때 삼항연산자를 쓰지 말자라는 룰을 정했습니다. 이유는 삼항연산자가 가독성이 떨어진다라고 판단했습니다. If else의 경우 길이는 길어지지만 삼항연산자보다 가독성이 높다 생각합니다!
그리고 해당 코드는 말씀해주신대로 isSelected를 적용하여 수정하였습니다!

Jeeehee replied:
삼항연산자가 가독성이 떨어진다고 생각하시는 이유를 알 수 있을까요??

HansolWorld replied:
지금은 변수명이 짧아서 가로로 읽는게 가독성이 좋지만, 변수명과 가로로 적어야할 내용이 길어진다면 가로로 읽는것이 가독성이 떨어진다 생각합니다. 위에서 아래로 읽어가는게 더 익숙한 방식이라 생각해서 삼항연산자를 쓰지말자라는 그라운드룰을 정했습니다.

Jeeehee replied:
그렇게 생각할수도 있겠네요!
하지만 위 코드만 봐도 if문을 사용하면 6줄을 읽어야 하는 것과 달리, 삼항연산자는 3줄의 코드만 보면 됩니다.
조건에 대한 처리를 한 줄로 정리할 수 있죠!
여러 조건에 대한 case를 나누어야 한다면, 사용하신 if문이나 switch 혹은 guard문을 사용해야겠지만
단순 비교에는 삼항연산자로 간결하게 처리하는 게 가독성이 좋을거라 생각합니다!

여기까지는 제 생각이고, 해당 부분은 두 분의 생각이 담겨있으니 수정하지 않으셔도 됩니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueFilter/IssueFilterHeaderCell.swift:
```diff
@@ -0,0 +1,48 @@
+
+import UIKit
+
+final class IssueFilterHeaderCell: UICollectionReusableView {
```
해당 뷰는 Cell이 아닌데, Naming을 Cell로 하신 이유가 있을까요?

Aiden0704 replied:
수정하였습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueFilter/IssueFilterViewController.swift:
```diff
@@ -0,0 +1,124 @@
+
+final class IssueFilterViewController: UIViewController {
+    let issueFilterHeaders = MockHeaderData()
+    
```
위, 아래 상수 모두 private으로 선언하지 않은 이유가 있을까요?

Aiden0704 replied:
실수입니다! 수정하였습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueFilter/IssueFilterViewController.swift:
```diff
@@ -0,0 +1,124 @@
+            self.collectionView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            self.collectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+        ])
+    }
```
Layout을 잡아주는 메서드면, configure 보단 `layoutCollectionView`가 더 낫지 않을까요??
configureCollectionView 메서드에 setRegister + Color와 같은 CollectionView 설정들을 넣고 Layout만 따로 빼는것도 괜찮을 것 같습니다!

Aiden0704 replied:
네이밍에 대해서 한번 더 생각해보겠습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueFilter/IssueFilterViewController.swift:
```diff
@@ -0,0 +1,124 @@
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, referenceSizeForHeaderInSection section: Int) -> CGSize {
+        let width = collectionView.bounds.width
+        let height: CGFloat = 44
```
UICollectionViewDelegateFlowLayout 첫번째 메서드에도 같은 값이 들어가 있는데, 이런 경우 공통 변수로 선언 후 사용하면 좋을것같아요!

Aiden0704 replied:
section에 관한것과 cell에 관한거라 서로 다른 설정이라 생각합니다
지금은 값이 같지만 다르게 설정될 여지가 있어 공통의 변수로 선언한다면 따로관리하기 어렵지 않을까 생각했습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueFilter/MockHeaderData.swift:
```diff
@@ -0,0 +1,12 @@
+import Foundation
+
+struct MockHeaderData {
+    private(set) var title: [String] = ["상태", "담당자", "레이블", "마일스톤"]
```
`private(set) var` 로 선언한 이유가 궁금합니다!

Aiden0704 replied:
실수입니다! 수정하겠습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueList/IssueCardCell.swift:
```diff
@@ -0,0 +1,127 @@
+
+final class IssueCardCell: UICollectionViewCell {
+    static let identifier = "Cell"
+    
```
역시나 모든 View 선언을 `private(set)`으로 하신 이유가 궁금합니다!
++ View Naming시 `IssueFilterHeaderCell`의 `titleLabel` 처럼, 뒤에 어떤 프로퍼티인지 적어주면 가독성이 더 좋아집니다.

Aiden0704 replied:
감사합니다! 네이밍에 대해서 한번 더 생각해보겠습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueList/IssueCardCell.swift:
```diff
@@ -0,0 +1,127 @@
+        ])
+    }
+    
+    private func setLabel() {
```
✔️ 해당 피드백은 모든 View에 적용해주세요!
아래와 같이 View의 초기 설정 코드는 View 선언시 함께 작성해두세요!
모든 설정을 한 곳에서 볼 수 있어, 추후 수정시 여러 파일 혹은 여러 메서드를 건들이지 않아도 됩니다.

Ex.
 ```swift
private let label: CustomLabelView = {
      let badgeView = CustomLabelView(text: "레이블", alignment: .left, font: FontStyle.body.font)
      badgeView.translatesAutoresizingMaskIntoConstraints = false
      badgeView.backgroundColor = .blue
      badgeView.textColor = .white
      badgeView.layer.cornerRadius = 12
      badgeView.clipsToBounds = true
      badgeView.textAlignment = .center
      return badgeView
}()
```

Aiden0704 replied:
감사합니다! 수정하겠습니다!

HansolWorld replied:
translatesAutoresizingMaskIntoConstraints = false의 경우 Layout에 관한 설정이라 생각하는데 layout 메서드가 아니라 초기 설정에 해주는게 맞을까요?

Jeeehee replied:
Layout에 관한 설정도 맞고, View에 관한 설정도 맞아서 이 부분은 편하신대로 하면 될것같네요! ㅎ_ㅎ

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueList/IssueListHeaderView.swift:
```diff
@@ -0,0 +1,32 @@
+final class IssueListHeaderView: UICollectionReusableView {
+    static let identifier = "HeaderView"
+    
+    private(set) var title: UILabel = {
```
아래 메서드를 통해 외부 접근 및 title을 변경해주고 있는데, 해당 변수를 private(set)으로 선언한 이유가 있을까요?

Aiden0704 replied:
실수입니다! 수정하겠습니다! 한번 더 확인하고 PR보내겠습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/IssueList/IssueViewController.swift:
```diff
@@ -0,0 +1,95 @@
+    }
+}
+
+extension IssueViewController: UICollectionViewDelegate {
```
사용하지 않는 코드는 PR을 올리실때 지우고 올려주세요 ~

Aiden0704 replied:
넵! 알겠습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/TabBar/TabBarViewController.swift:
```diff
@@ -0,0 +1,46 @@
+
+import UIKit
+
+final class TabBarViewController: UITabBarController {
```
역시나 하위 ViewController들을 private(set)으로 설정하신 이유가 있을까요?

Aiden0704 replied:
실수입니다! 수정하겠습니다!

Jeeehee commented on iOS/issue-tracker/issue-tracker/CommonTypeValue/FontStyle.swift:
```diff
@@ -0,0 +1,83 @@
+      
+struct Factor {
+    let font: UIFont
+    let lineLength: CGFloat
```
lineLength는 사용되지 않는 것 같은데 왜 선언해두신걸까요?

Aiden0704 replied:
lineLegth를 줘서 view의 height설정을 하고 싶었습니다. label.frame.size.height = ~~~.lineLength와 같은 방법으로 height를 변경을 하려 시도했는데 constraint가 변경되지 않아 일단은 lineLength를 남겨둔 채로 constraint에서 height를 변경해주는 방법으로 했습니다.

Jeeehee replied:
말씀하신 View가 UILabel이라면, `sizeToFit()` 과 [이 링크](https://babbab2.tistory.com/135)를 학습해 보시면 좋을 것 같아요!

Jeeehee commented:
질문하신 것에 대한 답입니다~

### 1. 어디까지 타입으로 묶어야 하는가?
말씀하신 Color, Font, Assets 은 타입으로 정의해 놓으면 적용할때도 편할뿐더러 추후 있을 수정에 빠른 대처가 가능하죠!
저는 여기에 휴먼 에러를 발생시킬 수 있는 것들 혹은 명시적으로 수치가 정해져 있는 것들은 타입으로 묶어 관리하고 있습니다.
(PlaceHolder나 Title, 범용적으로 쓰이는 Constraint 등)

### 2. Modal에서 NavigationBar 띄우기
우선 Modal과 NavigationContoller 그리고 iOS의 View 전환 방식을 학습해보시면 좋을 것 같습니다!
Modal은 Stack의 구조로 View가 쌓이는 NavigationController와 달리, 해당 화면에서 추가적인 작업이 있을때 현재 View 위에 띄우는 방식을 말합니다.
NavigationController Stack에 쌓이거나, 같은 흐름 선상에 놓이지 않습니다!
그러므로 잘 구현하셨습니다! 🙂

### 3. 뷰에서도 할 수 있다면, 인터페이스로 추상화를 하는게 일반적인가?
말씀하신 인터페이스가 CustomLabelView 가 맞을까요?
질문에 대한 답은 답변 주시면 하겠습니다!

### 4. StackView를 써야하는가?
Figma에 “설명을 최대 두 줄까지 보여줄 수 있다.” 라고 되어있는데
각각의 Label이 존재하는 게 아니라, ‘설명이 길다면 최대 2줄까지만 보여줘라’ 인것 같네요.
이런 경우엔 numberOfLines으로 줄 수 만 정해주면 됩니다!
AutoLayout을 잘 잡아두셨다면, 설명이 두 줄이 되더라도 문제가 없을테니까요.

그리고, 질문하신 것에 대한 답변은
Label이 쌓여야 할때는 StackView를 사용하기도 하고, Cell 안에 TableView를 넣기도 하고, Label 단위로 넣기도 합니다!
선호되는 방식은 정해져있지 않으며, 상황에 맞게 사용하게 됩니다!
그 상황을 판단하고 어떻게 구현할지는 개발자의 재량이라고 생각합니다!

그래서 결론은? 답이 없다~

Aiden0704 committed:
refactor: 2주차 PR 피드백 반영

- 메소드 name 변경
- 메소드 분리
- 페어와 코드 스타일 통합
- 불필요한 private(set) -> private 변경
- SwiftLint 수정

Jeeehee commented:
++ 혹시 TableView가 아닌 CollectionView로 구현하신 이유가 있을까요?

HansolWorld commented:
> ++ 혹시 TableView가 아닌 CollectionView로 구현하신 이유가 있을까요?

세로스크롤만 있는 상태에서 collectionview를 사용할 이유는 없지만, 이전 프로젝트에서 collectionView를 사용해서 좀더 익숙해지자라는 의미에서 collectionview를 선택했습니다. 금요일 jk시간에 스와이프가 tableview에서만 된다는 것을 알았고, 이 사실을 먼저 알았다면 tableView로 선택했을거 같습니다.

Jeeehee commented:
> > ++ 혹시 TableView가 아닌 CollectionView로 구현하신 이유가 있을까요?
> 
> 세로스크롤만 있는 상태에서 collectionview를 사용할 이유는 없지만, 이전 프로젝트에서 collectionView를 사용해서 좀더 익숙해지자라는 의미에서 collectionview를 선택했습니다. 금요일 jk시간에 스와이프가 tableview에서만 된다는 것을 알았고, 이 사실을 먼저 알았다면 tableView로 선택했을거 같습니다.

`UICollectionLayoutListConfiguration` 을 사용하면 Swipe Action을 구현할 수 있는데, 저도 구현해보지 않아서 어느정도의 난이도인지는 🥲
TableView로 변경하는 것도 방법이겠네요!

Jeeehee commented:
해당 PR은 머지하겠습니다! 두 분 이번주도 고생많으셨습니다~
